test/extended/builds/pipeline.go

```diff
@@ -820,4 +822,117 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 			buildAndSwitch("blue")
 		})
 	})
+	g.Context("cleanup jenkins jobs", func() {
+		g.BeforeEach(func() {
+			setupJenkins()
+
+			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
+				ticker = jenkins.StartJenkinsMemoryTracking(oc, oc.Namespace())
+			}
+
+			g.By("waiting for builder service account")
+			err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
+				ticker.Stop()
+			}
+		})
+
+		g.It("should delete a job run when the associated build is deleted", func() {
+
+			buildNameToJenkinsBuildURIMap := make(map[string]string)
+
+			g.By(fmt.Sprintf("calling oc create -f %q", nodejsDeclarativePipelinePath))
+			err := oc.Run("create").Args("-f", nodejsDeclarativePipelinePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting and canceling 5 pipeline builds")
+			for i := 1; i <= 5; i++ {
+				// start the build
+				_, _, _ = exutil.StartBuild(oc, "nodejs-sample-pipeline")
+
+				// get the build information
+				build, err := oc.BuildClient().Build().Builds(oc.Namespace()).Get(fmt.Sprintf("nodejs-sample-pipeline-%d", i), metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				buildNameToJenkinsBuildURIMap[build.Name] = build.Annotations[buildapi.BuildJenkinsBuildURIAnnotation]
+
+				// wait until the associated job run shows up in jenkins
+				err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+					_, status, err := j.GetResource(buildNameToJenkinsBuildURIMap[build.Name])
+					if err != nil {
+						fmt.Fprintf(g.GinkgoWriter, "%v", err)
+						return false, err
+					}
+					fmt.Fprintf(g.GinkgoWriter, "Verifying run for build %s. Status: %v ", build.Name, status)
+					if status != http.StatusOK {
+						fmt.Fprintf(g.GinkgoWriter, "Retrying: %v, %v", status, http.StatusOK)
+						return false, nil
+					}
+					return true, nil
+				})
```

Comment:
if the `j.GetResource` in the poll never succeeded after 1 minute, i.e. returned false/nil or false/err, are we sure there will be a run to delete below?

shouldn't we abort the test if we could never access the job/run log?